rslint_core/src/groups/errors/use_isnan.rs

```diff
@@ -0,0 +1,221 @@
+    use `isNaN(/* num */)` instead to test if a value is `NaN`. This rule is aimed at removing this footgun.
+
+    ## Invalid Code Examples
+
+    ```ignore
+    if (foo == NaN) {
+        // unreachable
+    }
+
+    if (NaN != NaN) {
+        // always runs
+    }
+    ```
+
+    ## Correct Code Examples
+
+    ```ignore
+    if (isNaN(foo)) {
+        /* */
+    }
+
+    if (!isNaN(foo)) {
+        /* */
+    }
+    ```
+    */
+    #[serde(default)]
+    UseIsnan,
+    errors,
+    "use-isnan",
+    /// Switch statements use `===` internally to match an expression, therefore `switch (NaN)` and `case NaN` will never match.
+    /// This rule disables uses like that which are always incorrect (true by default)
+    pub enforce_for_switch_case: bool,
+    /// Index functions like `indexOf` and `lastIndexOf` use `===` internally, therefore matching them against `NaN` will always
+    /// yield `-1`. This option disallows using `indexOf(NaN)` and `lastIndexOf(NaN)` (false by default)
+    pub enforce_for_index_of: bool
+}
+
+impl Default for UseIsnan {
+    fn default() -> Self {
+        Self {
+            enforce_for_switch_case: true,
+            enforce_for_index_of: false,
+        }
+    }
+}
+
+#[typetag::serde]
+impl CstRule for UseIsnan {
+    fn check_node(&self, node: &SyntaxNode, ctx: &mut RuleCtx) -> Option<()> {
+        match node.kind() {
+            BIN_EXPR => {
+                let expr = node.to::<BinExpr>();
+                if !expr.comparison() {
+                    return None;
+                }
+
+                let opposite = if expr.lhs().filter(|e| e.text() == "NaN").is_some() {
+                    expr.rhs()?
+                } else if expr.rhs().filter(|e| e.text() == "NaN").is_some() {
+                    expr.lhs()?
+                } else {
+                    return None;
+                };
+                let op = expr.op().unwrap();
+
+                let always_text = if matches!(op, op!(!=) | op!(!==)) {
+                    "true"
+                } else {
+                    "false"
+                };
+
+                let mut err = ctx
+                    .err(
+                        self.name(),
+                        format!(
+                            "comparing `{}` to `NaN` using `{}` will always return {}",
+                            opposite.text(),
+                            expr.op_token().unwrap().text(),
+                            always_text
+                        ),
+                    )
+                    .primary(expr.range(), "")
+                    .note("note: `NaN` is not equal to anything including itself");
+
+                // telling the user to use isNaN for `<`, `>`, etc is a bit misleading so we won't do it if that is the case
+                if op == op!(==) || op == op!(===) {
+                    err = err.note(format!(
+                        "help: use `isNaN` instead: `{}`",
+                        color(&format!("isNaN({})", opposite))
```

Comment:
Same as above

Reply:
I think this is fine, notes can have bigger text just fine